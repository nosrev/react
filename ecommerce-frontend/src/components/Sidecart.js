import React from 'react';
import { ReactComponent as Close } from '../svg/close.svg';
import image from '../img/produto.png';

const Sidecart = (props) => {
    return (
        <div id="side-cart" className={`${props.className} bg-white`} data-testid="side-cart">
            <button className="close-cart btn-none" onClick={props.closeCart}>
                <Close className="icon w-5 stroke-black stroke-2"/>
            </button>
            <h3 className="font-semibold text-2xl text-center border-b py-2 px-5">Cart</h3>

            <div className="products mt-2 px-3 overflow-auto h-3/4">
                <div className="product flex border-b py-3">
                    <img src={image} className="w-20 mr-3" alt="Produto" />
                    <div>
                        <h6 className="font-medium">Flecked cotton-blend</h6>
                        <span className="font-medium text-lg text-primary">$175.00</span>
                    </div>
                </div>
                <div className="product flex border-b py-3">
                    <img src={image} className="w-20 mr-3" alt="Produto" />
                    <div>
                        <h6 className="font-medium">Flecked cotton-blend</h6>
                        <span className="font-medium text-lg text-primary">$175.00</span>
                    </div>
                </div>
                <div className="product flex border-b py-3">
                    <img src={image} className="w-20 mr-3" alt="Produto" />
                    <div>
                        <h6 className="font-medium">Flecked cotton-blend</h6>
                        <span className="font-medium text-lg text-primary">$175.00</span>
                    </div>
                </div>
                <div className="product flex border-b py-3">
                    <img src={image} className="w-20 mr-3" alt="Produto" />
                    <div>
                        <h6 className="font-medium">Flecked cotton-blend</h6>
                        <span className="font-medium text-lg text-primary">$175.00</span>
                    </div>
                </div>
                <div className="product flex border-b py-3">
                    <img src={image} className="w-20 mr-3" alt="Produto" />
                    <div>
                        <h6 className="font-medium">Flecked cotton-blend</h6>
                        <span className="font-medium text-lg text-primary">$175.00</span>
                    </div>
                </div>
            </div>

            <div className="absolute bottom-4 w-full px-3">
                <a href="/" className="border bg-white border-primary w-full block text-center text-primary py-1">
                    Carrinho
                </a>
                <a href="/" className="border bg-primary border-primary w-full block text-white text-center py-1 mt-2">
                    Checkout
                </a>
            </div>
        </div>
    )
}

export default Sidecart;
