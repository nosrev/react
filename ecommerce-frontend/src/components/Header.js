import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import logo from '../img/logo.png';
import { ReactComponent as Bag } from '../svg/bag.svg';
import { ReactComponent as Search } from '../svg/search.svg';
import { ReactComponent as Close } from '../svg/close.svg';
import Sidecart from './Sidecart';

const Header = () => {
    const [showSearch, setShowSearch] = useState("");
    const [showCart, setShowCart] = useState(false);
    const [showMenu, setShowMenu] = useState("");

    const closeCart = () => {
      setShowCart((previousState) => !previousState)
    };

    const openMenu = () => {
        if (showMenu) {
            setShowMenu("");
            document.body.classList.remove('cbp-spmenu-push-toright');
            document.querySelector(".cbp-spmenu").classList.remove('cbp-spmenu-open');
        } else {
            setShowMenu("active");
            document.body.classList.add('cbp-spmenu-push-toright');
            document.querySelector(".cbp-spmenu").classList.add('cbp-spmenu-open');
        }
    };

    return (
        <div id="header">
            <div className="container mx-auto flex flex-row justify-between items-center">
                <div className="flex">
                    <button id="showMenu" onClick={() => openMenu()} className={ showMenu }>
                        <span aria-hidden="true"></span>
                        <span aria-hidden="true"></span>
                        <span aria-hidden="true"></span>
                    </button>
                    <Link to='/'>
                        <img src={logo} alt="Store" />
                    </Link>
                </div>
                <ul className="menu list-none">
                    <li>
                        <Link to='/'>
                            Home
                        </Link>
                    </li>
                    <li>
                        <Link to='/shop'>
                            Loja
                        </Link>
                    </li>
                    <li><a href="/">Blog</a></li>
                    <li><a href="/">Contato</a></li>
                </ul>

                <div className="flex items-center">
                    <ul className="links list-none">
                        <li><a href="/">Entrar</a></li>
                        <li><a href="/">Cadastrar</a></li>
                    </ul>

                    <div className="flex items-center">
                        <button className="cart-btn" onClick={() => setShowCart(!showCart)}>
                            <Bag className="icon w-10 fill-white" />
                            <span className="counter font-bold bg-white px-2 py-1 rounded-full">0</span>
                        </button>
                        <button id="toggle-search" onClick={() => setShowSearch("show")}>
                            <Search className="icon w-8 fill-white" />
                        </button>
                    </div>
                </div>

                <div className="cbp-spmenu cbp-spmenu-vertical cbp-spmenu-left" id="menu-mobile">
                    <ul>
                        <li>
                            <Link to='/'>
                                Home
                            </Link>
                        </li>
                        <li><a href="/">Entrar</a></li>
                        <li><a href="/">Cadastrar</a></li>
                        <li>
                            <Link to='/shop'>
                                Loja
                            </Link>
                        </li>
                        <li><a href="/">Blog</a></li>
                        <li><a href="/">Contato</a></li>
                    </ul>
                </div>
            </div>

            <div id="search-form" className={ showSearch }>
                <button className="close btn-none" onClick={() => setShowSearch("")}>
                    <Close className="icon w-10 stroke-white stroke-2" />
                </button>

                <div className="flex items-center">
                    <input type="text" className="input" placeholder="Digite o que deseja" />
                    <button className="btn-none" id="btn-search">
                        <Search className="icon w-8" />
                    </button>
                </div>
            </div>

            <Sidecart className={ showCart ? "open" : "" } closeCart={closeCart} />
        </div>
    );
}

export default Header;
