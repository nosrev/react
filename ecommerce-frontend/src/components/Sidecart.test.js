import {render, screen, fireEvent} from '@testing-library/react'
import renderer from 'react-test-renderer';
import Sidecart from './Sidecart';
import Header from './Header';
import { MemoryRouter as Router } from 'react-router-dom';

it('render Sidecart', () => {
  const component = renderer.create(
    <Sidecart />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("test callback function", async () => {
    const { container } = render(<Router><Header /></Router>)
    const element = screen.getByTestId('side-cart');
    fireEvent.click(await container.querySelector('.cart-btn'));
    expect(element.classList.contains('open')).toBe(true)

    fireEvent.click(await container.querySelector('.close-cart'));
    expect(element.classList.contains('open')).toBe(false)

    fireEvent.click(await container.querySelector('.cart-btn'));
    expect(element.classList.contains('open')).toBe(true)

    fireEvent.click(await container.querySelector('.cart-btn'));
    expect(element.classList.contains('open')).toBe(false)
});
