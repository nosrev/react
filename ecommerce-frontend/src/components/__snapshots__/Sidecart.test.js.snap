// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Sidecart 1`] = `
<div
  className="undefined bg-white"
  data-testid="side-cart"
  id="side-cart"
>
  <button
    className="close-cart btn-none"
  >
    <svg
      className="icon w-5 stroke-black stroke-2"
    >
      close.svg
    </svg>
  </button>
  <h3
    className="font-semibold text-2xl text-center border-b py-2 px-5"
  >
    Cart
  </h3>
  <div
    className="products mt-2 px-3 overflow-auto h-3/4"
  >
    <div
      className="product flex border-b py-3"
    >
      <img
        alt="Produto"
        className="w-20 mr-3"
        src="produto.png"
      />
      <div>
        <h6
          className="font-medium"
        >
          Flecked cotton-blend
        </h6>
        <span
          className="font-medium text-lg text-primary"
        >
          $175.00
        </span>
      </div>
    </div>
    <div
      className="product flex border-b py-3"
    >
      <img
        alt="Produto"
        className="w-20 mr-3"
        src="produto.png"
      />
      <div>
        <h6
          className="font-medium"
        >
          Flecked cotton-blend
        </h6>
        <span
          className="font-medium text-lg text-primary"
        >
          $175.00
        </span>
      </div>
    </div>
    <div
      className="product flex border-b py-3"
    >
      <img
        alt="Produto"
        className="w-20 mr-3"
        src="produto.png"
      />
      <div>
        <h6
          className="font-medium"
        >
          Flecked cotton-blend
        </h6>
        <span
          className="font-medium text-lg text-primary"
        >
          $175.00
        </span>
      </div>
    </div>
    <div
      className="product flex border-b py-3"
    >
      <img
        alt="Produto"
        className="w-20 mr-3"
        src="produto.png"
      />
      <div>
        <h6
          className="font-medium"
        >
          Flecked cotton-blend
        </h6>
        <span
          className="font-medium text-lg text-primary"
        >
          $175.00
        </span>
      </div>
    </div>
    <div
      className="product flex border-b py-3"
    >
      <img
        alt="Produto"
        className="w-20 mr-3"
        src="produto.png"
      />
      <div>
        <h6
          className="font-medium"
        >
          Flecked cotton-blend
        </h6>
        <span
          className="font-medium text-lg text-primary"
        >
          $175.00
        </span>
      </div>
    </div>
  </div>
  <div
    className="absolute bottom-4 w-full px-3"
  >
    <a
      className="border bg-white border-primary w-full block text-center text-primary py-1"
      href="/"
    >
      Carrinho
    </a>
    <a
      className="border bg-primary border-primary w-full block text-white text-center py-1 mt-2"
      href="/"
    >
      Checkout
    </a>
  </div>
</div>
`;
